From: Marco Martin <notmart@gmail.com>
Date: Tue, 17 Jan 2017 17:23:07 +0100
Subject: correctly forward status from applets to containment

Summary:
unfortunately hiddenvisibility is the "biggest" status
that makes computing the proper status tricky.
if an applet in the containment is hidden, we must not mark as
hidden the whole containment, because it's as if hidden was
(as it should be) the smallest in the enum

BUG:372062

Test Plan: touchpad plasmoid no longer hides the systray

Reviewers: #plasma, davidedmundson

Reviewed By: #plasma, davidedmundson

Subscribers: plasma-devel, #frameworks

Tags: #plasma, #frameworks

Differential Revision: https://phabricator.kde.org/D4174
---
 src/plasma/plasma.h                  | 1 +
 src/plasma/private/containment_p.cpp | 4 +++-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/plasma/plasma.h b/src/plasma/plasma.h
index 523e32915..eb34aee3f 100644
--- a/src/plasma/plasma.h
+++ b/src/plasma/plasma.h
@@ -255,6 +255,7 @@ public:
         NeedsAttentionStatus = 3, /**< The Item needs attention **/
         RequiresAttentionStatus = 4, /**< The Item needs persistent attention **/
         AcceptingInputStatus = 5, /**< The Item is accepting input **/
+        //FIXME KF6: this should be the smallest status
         HiddenStatus = 6    /**< The Item will be hidden totally  **/
     };
     Q_ENUMS(ItemStatus)
diff --git a/src/plasma/private/containment_p.cpp b/src/plasma/private/containment_p.cpp
index 061176b95..646c80021 100644
--- a/src/plasma/private/containment_p.cpp
+++ b/src/plasma/private/containment_p.cpp
@@ -123,7 +123,9 @@ void ContainmentPrivate::checkStatus(Plasma::Types::ItemStatus appletStatus)
         }
     }
 
-    q->setStatus(appletStatus);
+    if (appletStatus > q->status() && appletStatus != Plasma::Types::HiddenStatus) {
+        q->setStatus(appletStatus);
+    }
 }
 
 void ContainmentPrivate::triggerShowAddWidgets()
